### PR TITLE
chore: Make use of statefulMap instead.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
Make use of statefulMap

### CODE REVIEW

File: google-common/src/main/scala/org/apache/pekko/stream/connectors/google/util/AnnotateLast.scala [hunk 1]
```diff
@@ -47,18 +47,13 @@ private[google] object AnnotateLast {
 
   def apply[T]: Flow[T, MaybeLast[T], NotUsed] =
     Flow[T]
-      .map(Some(_))
-      .concat(Source.single(None))
```
Comment (He-Pin) [inline google-common/src/main/scala/org/apache/pekko/stream/connectors/google/util/AnnotateLast.scala:51]: some kind of `fs`'s `noneTerminate` operator.

**Review by pjfanning — APPROVED** [review]
lgtm - but can't be merged util we set up the 1.1 build - basically #325 plus a tag that I will add after 325 is merged

**Comment by pjfanning** [comment]
v1.1 commits are now allowed

**Comment by He-Pin** [comment]
@pjfanning @mdedetrich can this be merged now？

**Review by mdedetrich — APPROVED** [review]
lgtm and yes it can since we are on 1.1.0 now

**Pull request merged**